Content.Server/_Goobstation/Devil/DevilSystem.Handshake.cs

```diff
@@ -0,0 +1,120 @@
+    {
+        SubscribeLocalEvent<DevilComponent, GetVerbsEvent<InnateVerb>>(OnGetVerbs);
+        SubscribeLocalEvent<PendingHandshakeComponent, GetVerbsEvent<InnateVerb>>(OnGetVerbsPending);
+    }
+    private void OnGetVerbs(EntityUid uid, DevilComponent comp, GetVerbsEvent<InnateVerb> args)
+    {
+        // Can't shake your own hand, and you can't shake from a distance
+        if (!args.CanAccess
+        || !args.CanInteract
+        || _state.IsIncapacitated(args.Target)
+        || !HasComp<MobStateComponent>(args.Target)
+        || HasComp<CondemnedComponent>(args.Target)
+        || args.Target == args.User
+        || !_body.BodyHasPartType(uid, BodyPartType.Hand) // cant shake if you have no hands
+        || !_body.BodyHasPartType(args.Target, BodyPartType.Hand) // or if they have none
+        || !_contract.IsUserValid(args.Target, out _))
+            return;
+
+        InnateVerb handshakeVerb = new()
+        {
+            Act = () => OfferHandshake(args.User, args.Target),
+            Text = Loc.GetString("hand-shake-prompt-verb", ("target", args.Target)),
+            Icon = new SpriteSpecifier.Rsi(new("_Goobstation/Actions/devil.rsi"), "summon-contract"),
+            Priority = 1 // Higher priority than default verbs
+        };
+        args.Verbs.Add(handshakeVerb);
+    }
+
+    private void OnGetVerbsPending(EntityUid uid, PendingHandshakeComponent comp, GetVerbsEvent<InnateVerb> args)
+    {
+        if (!args.CanAccess
+            || !args.CanInteract
+            || _state.IsIncapacitated(args.Target)
+            || !HasComp<MobStateComponent>(args.Target)
+            || args.Target != comp.Offerer)
+            return;
+
+        InnateVerb handshakeVerb = new()
+        {
+            Act = () => HandleHandshake(args.Target, args.User),
+            Text = Loc.GetString("hand-shake-accept-verb", ("target", args.Target)),
+            Icon = new SpriteSpecifier.Rsi(new("_Goobstation/Actions/devil.rsi"), "summon-contract"),
+            Priority = 1 // Higher priority than default verbs
+        };
+        args.Verbs.Add(handshakeVerb);
+    }
+
+    private void OfferHandshake(EntityUid user, EntityUid target)
+    {
+        if (HasComp<DevilComponent>(target)
+            || HasComp<PendingHandshakeComponent>(target)
+            || !_contract.IsUserValid(target, out _))
+            return;
+
+        var pending = AddComp<PendingHandshakeComponent>(target);
+        pending.Offerer = user;
+        pending.ExpiryTime = _timing.CurTime + TimeSpan.FromSeconds(15);
+
+        // Notify target
+        var popupMessage = Loc.GetString("handshake-offer-popup", ("user", user));
+        _popup.PopupEntity(popupMessage, target, target);
+
+        // Notify self
+        var selfPopup = Loc.GetString("handshake-offer-popup-self", ("target", target));
+        _popup.PopupEntity(selfPopup, user, user);
+    }
+    private void HandleHandshake(EntityUid user, EntityUid target)
```

Comment:
Blank line between methods
```suggestion
    }

    private void HandleHandshake(EntityUid user, EntityUid target)
```